src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -197,9 +197,17 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
             final boolean isSuccessful = done.await(settings.getKafkaSendTimeoutMs(), TimeUnit.MILLISECONDS);
 
             if (!isSuccessful) {
+                for (final BatchItem item : batch) {
+                    item.safeFail("timed out");
```

Comment:
`isSuccessful` will be false in case the submit operation exceeded the time specified by the config.

But when it happens there could be some successfully published items already. So I try to fail every item.

There could be race conditions between this update and the `callback` updates, so I used the `synchronized` keyword.